src/main/groovy/org/boozallen/plugins/jte/init/primitives/injectors/StageInjector.groovy

```diff
@@ -34,12 +38,42 @@ import org.jenkinsci.plugins.workflow.flow.FlowExecutionOwner
     }
 
     @Override
-    void doInject(FlowExecutionOwner flowOwner, PipelineConfigurationObject config, Binding binding){
+    @RunAfter([LibraryStepInjector, DefaultStepInjector, TemplateMethodInjector])
+    void injectPrimitives(FlowExecutionOwner flowOwner, PipelineConfigurationObject config, TemplateBinding binding){
         Class stageClass = getPrimitiveClass()
         config.getConfig().stages.each{ name, steps ->
-            ArrayList<String> stepsList = []
-            steps.collect(stepsList){ step -> step.key }
-            binding.setVariable(name, stageClass.newInstance(binding, name, stepsList))
+            List<String> stepNames = steps.keySet() as List<String>
+            binding.setVariable(name, stageClass.newInstance(binding, name, stepNames))
+        }
+    }
+
+    @Override
+    void validateBinding(FlowExecutionOwner flowOwner, PipelineConfigurationObject config, TemplateBinding binding){
+        LinkedHashMap aggregatedConfig = config.getConfig()
+        LinkedHashMap stagesWithUndefinedSteps = [:]
+        aggregatedConfig.stages.each{ name, stageConfig ->
+            List<String> steps = stageConfig.keySet() as List<String>
+            List<String> undefinedSteps = []
+            steps.each{ step ->
+                if(!binding.hasStep(step)){
+                    undefinedSteps << step
+                }
+            }
+            if(undefinedSteps){
+                stagesWithUndefinedSteps[name] = undefinedSteps
+            }
+        }
+        if(stagesWithUndefinedSteps){
+            List<String> error = [
+                    "There are Stages defined that reference steps that do not exist.",
+                    "Consider adding step names to the template_methods block"
```

Comment:
should this become "Consider adding step names to the template_methods block or check your libraries" or some other verbiage that implies to check the configured libraries